python/lsst/afw/image/_exposureSummaryStats.py

```diff
@@ -20,76 +20,102 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 from __future__ import annotations
 
-from dataclasses import dataclass, asdict, field
-from typing import List
+import dataclasses
+from typing import TYPE_CHECKING
 import yaml
 import warnings
 
 from ..typehandling import Storable, StorableHelperFactory
 
+if TYPE_CHECKING:
+    from ..table import BaseRecord, Schema
```

Comment:
I have no doubt you've got this right, but I don't know what determines the state of `TYPE_CHECKING` and am wondering what happens if it's `False` and these imports don't happen?

Reply:
This is only set when running mypy or something similar to do type-checking, so in this case (i.e. in afw) that amounts to "just inside some people's editors", and certainly never in production code; you'll note that these symbols are never used anywhere other than type annotations, and that means at runtime they aren't actually needed (and when they are, I've imported them inside function bodies).  That's all because afw subpackages are mess of circular dependencies and I wasn't at all sure that importing `lsst.afw.table` at module scope normally was safe here.

Reply:
Thanks for the explanation!